doc/making-a-visualizer.md

```diff
@@ -1,17 +1,194 @@
 
-Big picture, every visualizer needs to implement the interface defined in
-`VisualizerInterface.ts` which specifies the basic expectations of a
-visualizer. (See below for the easy way to do this.) This interface includes
-the following data and methods:
+### Get started
+
+#### Choose a graphics framework
+
+The easiest way to build a visualizer is to extend a pre-made visualizer base
+class, which automatically sets up a graphics framework for you to use. Right
+now, there's only one base class available:
+
+-   [`P5Visualizer`](#p5-visualizers) uses the
+    [**p5.js**](https://p5js.org/learn/) library for graphics and user
+    interaction.
+
+If you want to use a new graphics framework, you'll need to write your own
+implementation of the [visualizer interface](#abstract-visualizers).
+
+#### Test your visualizer on the workbench
+
+While you're working on a visualizer, we recommend keeping it in the
+`src/visualizers-workbench` directory, where Frontscope typically won't notice
+it. To load it, run Frontscope in "workbench mode" by calling
+`npm run dev:workbench`.
+
+When you're ready to propose your visualizer as an official part of
+Numberscope, you'll move it to `src/visualizers`. Then Frontscope will notice
+and load it when you call the usual `npm run dev`. For details, see
+[below](#where-to-put-your-visualizer).
+
+### p5 visualizers
+
+A good way to start a p5 visualizer is to copy and modify one of the basic
+examples in the `src/visualizers-workbench` directory:
+
+-   `P5VisualizerTemplate.ts`
+-   `Differences.ts`
+
+Let's look at the parts of a p5 visualizer. We recommend following along in
+one of the basic examples as you read.
+
+#### :key: Name _(required)_
+
+Put the visualizer's name in the static `visualizationName` property. The name
+is used in the visualizer list and the titles of bundle cards.
+
+#### :bulb: Parameters _(often used)_
+
+Parameters are the user-facing structures that ask for values when a
+visualizer is created. When the user clicks "save", each parameter stores its
+value in a corresponding top-level property of the visualizer object. If you
+change one of those top-level properties while the visualizer is running, you
+should refresh the corresponding parameter by calling `this.refreshParams()`.
+
+Below the list of parameter properties, the visualizer class has a `params`
+property that describes how the parameters should appear in the UI. Look in
+`src/shared/Paramable.ts`, or in other visualizers, to learn about the options
+you can set in the `params` property.
+
+-   **p5 Visualizer Template:** `stepSize = 1`.
+-   **Differences:** `n = 20`, `levels = 5`. The parameters are refreshed in
+    the in the `inhabit()` method.
+
+#### :bulb: Other top-level properties _(often used)_
+
+You may also need top-level properties that are set and updated while the
+visualizer is running, beyond the user's direct control. By convention, we
+list these properties after the `params` property.
```

Comment:
you might add something like "For example,  you might want variables that keep track of the progress in drawing your visualization." before "By convention..." Or maybe it's fine as is. Just a thought, feel free to go either way.

Reply:
Aaron and I talked about this and he may just insert one concrete "for example" here.